src/shared/components/Button/Button.style.ts

```diff
@@ -72,53 +75,116 @@ const colorsFromProps = ({ variant }: ButtonStyleProps) => {
   return styles
 }
 
-const sizeFromProps = ({ size = 'regular', full, hasText }: ButtonStyleProps) => {
-  let padding, fontSize
+const sizeFromProps = ({ size = 'medium', full, hasText }: ButtonStyleProps) => {
+  let padding, fontSize, buttonSizeWithoutText
   switch (size) {
-    case 'smaller': {
-      padding = '10px'
+    case 'small': {
+      padding = '9.5px 12px'
       fontSize = typography.sizes.button.small
+      buttonSizeWithoutText = '32px'
       break
     }
-    case 'small': {
-      padding = hasText ? `12px 14px` : '12px'
+    case 'medium': {
+      padding = hasText ? `12px 16px` : '12px'
       fontSize = typography.sizes.button.medium
+      buttonSizeWithoutText = '40px'
       break
     }
-    case 'regular':
+    case 'large':
     default: {
-      padding = hasText ? `14px 20px` : '14px'
+      padding = hasText ? `15px 20px` : '14px'
       fontSize = typography.sizes.button.large
+      buttonSizeWithoutText = '48px'
       break
     }
   }
   return css`
-    width: ${full ? '100%' : ''};
+    width: ${!hasText ? buttonSizeWithoutText : full ? '100%' : ''};
+    height: ${!hasText ? buttonSizeWithoutText : 'auto'};
```

Comment:
Why do we need to set the height explicitly? I think we should be fine with just the padding?

Reply:
It's here to keep 1:1 ratio for buttons with just the icon